frontend/javascripts/oxalis/shaders/main_data_fragment.glsl.js

```diff
@@ -139,30 +138,30 @@ void main() {
     gl_FragColor = vec4(bucketPosition, zoomStep) / 255.;
     return;
   }
+  vec3 data_color = vec3(0.0);
 
-  <% if (hasSegmentation) { %>
-    vec4 id = vec4(0.);
-    vec4 cellIdUnderMouse = vec4(0.);
+  <% _.each(segmentationLayerNames, function(segmentationName, layerIndex) { %>
```

Comment:
> Doesn't doing it this way with the respective variables and functions increase the shader length and complexity, leading to longer compile times.

Yes, that's correct, in general (even though, often still acceptable probably). However:

>  Since we're only ever rendering a single segmentation layer, couldn't we leave the shader code mostly as-is and only swap the textures + introduce and change some uniforms (packingDegree, layerIndex, etc.)?

The mid-term vision for this feature is to support simultaneous rendering of multiple segmentation layers (this is also listed in the follow-up issue). Therefore, I already built the shader structure that way :)

Reply:
Ah, I should have paid better attention. Thanks for clarifying, all good then :)